core/src/main/java/org/elasticsearch/index/similarity/IBSimilarityProvider.java

```diff
@@ -65,7 +66,7 @@
 
     private final IBSimilarity similarity;
 
-    public IBSimilarityProvider(String name, Settings settings, Settings indexSettings) {
+    public IBSimilarityProvider(String name, Settings settings, Settings indexSettings, ScriptService scriptService) {
```

Comment:
Can we avoid adding ScriptService to the ctor of every provider? It should only be needed for the new one?